harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -320,28 +330,32 @@ def gpu_mem(self) -> int:
         assert (
             self.model_profile_info_trial is not None
         ), "The model profile info Trial must be run before calling this method."
-        return self.model_profile_info_trial.metric["gpu_mem"]
+        assert isinstance(self.model_profile_info_trial.metric, dict)
+        return self.model_profile_info_trial.metric.get("gpu_mem", 0)
 
     @property
     def num_params(self) -> int:
```

Comment:
Similar statements to the above for `num_params`, `trainable_num_params`, and `activation_mem_per_gpu`: each should definitely be in a key in `self.model_profile_info_trial.metric` (which must be a dict) and if not, then we need to error out immediately.